### ISSUE TITLE
can't install neopixel

### ISSUE THREAD

**Issue by gtoal** [standupmaths/xmastree2020#32]
Following the instructions at https://learn.adafruit.com/neopixels-on-raspberry-pi/python-usage on how to install the
neopixel library, I get this:

```
Collecting sysv-ipc (from Adafruit-Blinka->adafruit-circuitpython-neopixel)
  Using cached https://files.pythonhosted.org/packages/0c/d7/5d2f861155e9749f981e6c58f2a482d3ab458bf8c35ae24d4b4d5899ebf9/sysv_ipc-1.1.0.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-p029qfz7/sysv-ipc/setup.py", line 11, in <module>
        import prober
      File "/tmp/pip-build-p029qfz7/sysv-ipc/prober.py", line 137
        d["SYSV_IPC_VERSION"] = f'"{version}"'
                                             ^
    SyntaxError: invalid syntax

```

I checked that I have python3 installed.

The second command doesn't work either:

```
pi@kano4:~/xmastree2020-main $ sudo python3 -m pip install --force-reinstall adafruit-blinka
Collecting adafruit-blinka
  Using cached https://www.piwheels.org/simple/adafruit-blinka/Adafruit_Blinka-5.13.1-py3-none-any.whl
Collecting sysv-ipc (from adafruit-blinka)
  Using cached https://files.pythonhosted.org/packages/0c/d7/5d2f861155e9749f981e6c58f2a482d3ab458bf8c35ae24d4b4d5899ebf9/sysv_ipc-1.1.0.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-5yeeu738/sysv-ipc/setup.py", line 11, in <module>
        import prober
      File "/tmp/pip-build-5yeeu738/sysv-ipc/prober.py", line 137
        d["SYSV_IPC_VERSION"] = f'"{version}"'
                                             ^
    SyntaxError: invalid syntax

```
Ignoring the error and trying to run anyway didn't work...

```
pi@kano4:~/xmastree2020-main $ python3 xmaslights-spin.py
Traceback (most recent call last):
  File "xmaslights-spin.py", line 144, in <module>
    xmaslight()
  File "xmaslights-spin.py", line 9, in xmaslight
    import neopixel
ImportError: No module named 'neopixel'
pi@kano4:~/xmastree2020-main $ python xmaslights-spin.py
Traceback (most recent call last):
  File "xmaslights-spin.py", line 144, in <module>
    xmaslight()
  File "xmaslights-spin.py", line 43, in xmaslight
    pixels = neopixel.NeoPixel(board.D18, PIXEL_COUNT, auto_write=False)
AttributeError: 'module' object has no attribute 'NeoPixel'

```

Any suggestions what to do next? (btw python newbie, so can I ask you to spell it out and not assume I know much about how python works)

thanks

Graham

**Comment by raxod502** [comment]
You need to upgrade to Python 3.6, this module is using f-strings which are not supported in Python <=3.5. https://realpython.com/python-f-strings/

**Comment by oliverdunk** [comment]
Hey @gtoal! I hope you made some progress here. I'm going to close this as we don't really have a use for the issues section right now, but feel free to continue the discussion. Thanks for jumping in to help @raxod502!

**Comment by gtoal** [comment]
Thanks, raxod502 - I didn't realise my submission had been answered until I got the notification just now about closing the issue.  I'll try up (down?) grading python and see what happens.
... 2 hrs later ...
That did the trick!  I did have to do some other tweaks such as making python, python3, and pip3 links in /usr/local/bin to the python3.6 installation.  I followed https://installvirtual.com/install-python-3-on-raspberry-pi-raspbian/ for the python3.6 upgrade.
